src/common/lockdep.cc

```diff
@@ -175,6 +180,15 @@ static int _lockdep_register(const char *name)
     }
     lock_ids[name] = id;
     lock_names[id] = name;
+    if (current_maxid + 1 >= follows.size()) {
+      follows.resize(current_maxid + 1);
+      follows_bt.resize(current_maxid + 1);
+    }
+    if (follows.size() == follows.capacity()) {
+      int new_size = std::min((int)(follows.size() * 2), MAX_LOCKS);
+      follows.reserve(new_size);
+      follows_bt.reserve(new_size);
+    }
```

Comment:
Is this really necessary?

Reply:
From my test this will affect the perf somehow, espically when testing the DirLs which may register many locks suddenly.

Reply:
i don't understand this. as libstdc++ double the size by default when the capacity of a vector is smaller than expected when resizing it. see https://code.woboq.org/gcc/libstdc++-v3/include/bits/stl_vector.h.html#1641

Reply:
Yeah, that works for inode lock feature.